wandb/sdk/wandb_run.py

```diff
@@ -1467,32 +1467,54 @@ def _datatypes_callback(self, fname: str) -> None:
         files: FilesDict = dict(files=[(GlobStr(glob.escape(fname)), "now")])
         self._backend.interface.publish_files(files)
 
-    def _serialize_custom_charts(self, data: dict[str, Any]) -> dict[str, Any]:
+    def _serialize_custom_charts(
+        self,
+        data: dict[str, Any],
+        key_prefix: str | None = None,
+    ) -> dict[str, Any]:
+        """Transform plot objects into tables, and adds the vega spec to the run config.
+
+        Args:
+            data: Dictionary containing data that may include plot objects
+                Plot objects can be nested in dictionaries, which will be processed recursively.
+
+        Returns:
+            The processed dictionary with custom charts transformed into tables.
+        """
```

Comment:
I agree it being modified in place is confusing. I have updated the code based on another comment you left regarding the table key. As part of that I modified this function now to return a new object that, rather than trying to modify and and move keys and values.